### PR TITLE
feat: inject GITHUB_SHA to deployments

### PR DESCRIPTION
**Pull request by nhamlh**
# Ticket
https://app.clickup.com/t/2575789/TECH-10692

Tested with:
- access-control (Golang): https://github.com/goes-funky/access-control/actions/runs/3232204015
- chat-backend (PHP): https://github.com/goes-funky/chat-backend/actions/runs/3232214347
- alerts-service (Python): https://github.com/goes-funky/alerts-service/actions/runs/3232352688

### CODE REVIEW

File: pkg/common/steps.cue [hunk 1]
```diff
@@ -135,4 +135,41 @@ package common
...
+//NOTE: Currently only add env to first container of pod
+#skaffold_kustomize_patch_yaml: """
+- op: add
+  path: "/spec/template/spec/containers/0/env/-"
+  value:
+    name: OTEL_RESOURCE_ATTRIBUTES
+    value: "service.version=$GITHUB_SHA"
+- op: add
+  path: "/spec/template/metadata/labels/tags.datadog.com~1service.version"
+  value: "$GITHUB_SHA"
```
Comment (Steffen911) [inline pkg/common/steps.cue:171]: @nhamlh Can we use the short-sha like we do in https://github.com/goes-funky/lineage-service/blob/main/skaffold.yaml#L28?

File: pkg/common/steps.cue [hunk 2]
```diff
@@ -135,4 +135,41 @@ package common
...
+						skaffold deploy --force --build-artifacts=build.json
+						"""
+		}
+	}
 }
+
+//NOTE: Currently only add env to first container of pod
+#skaffold_kustomize_patch_yaml: """
+- op: add
+  path: "/spec/template/spec/containers/0/env/-"
```
Comment (Steffen911) [inline pkg/common/steps.cue:165]: What happens if there are multiple containers in one setup? That will always update the first one only, right?

File: .github/workflows/deploy.yaml [hunk 3]
```diff
@@ -106,7 +106,31 @@ jobs:
...
+
+          # create patch file
+          cat >patches.yaml <<EOF
+          - op: add
+            path: "/spec/template/spec/containers/0/env/-"
+            value:
+              name: OTEL_RESOURCE_ATTRIBUTES
+              value: "service.version=$GITHUB_SHA"
+          - op: add
+            path: "/spec/template/metadata/labels/tags.datadog.com~1service.version"
```
Comment (Steffen911) [inline .github/workflows/deploy.yaml:120]: What is this ~1 doing?

**Review by Steffen911 — COMMENTED** [review]
My overall feeling is that this is quite a big hack just to avoid touching 20 repos. If something doesn't go well, because a yaml might not fit the expectations it will also be extremely hard to debug.

My preference is that we just apply the change to all services that are listed in Datadog. That should also be done within an hour.

What do you think @nhamlh @nils-borrmann-y42 ?

File: .github/workflows/deploy.yaml (hunk 3 above)
Comment (nhamlh) [reply .github/workflows/deploy.yaml:120]: It's rendered to `/`.

File: pkg/common/steps.cue (hunk 2 above)
Comment (nhamlh) [reply pkg/common/steps.cue:165]: Yes you're right, as with the note. That's current drawback of the patch.

File: pkg/common/steps.cue (hunk 1 above)
Comment (nhamlh) [reply pkg/common/steps.cue:171]: Yes!

**Comment by nhamlh** [comment]
@Steffen911 I agree with updating every repos. We might miss the setup with newer repo but I think we can easily catch that with written process or a skaffold template somewhere.

**Comment by nhamlh** [comment]
Closing & update skaffold with service repos instead.

**Pull request closed without merging**